provd/internal/services/hello/hello_test.go

```diff
@@ -0,0 +1,103 @@
+package hello_test
+
+import (
+	"context"
+	"flag"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/canonical/ubuntu-desktop-provision/provd"
+	"github.com/canonical/ubuntu-desktop-provision/provd/internal/services/hello"
+	"github.com/canonical/ubuntu-desktop-provision/provd/internal/testutils"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func TestEmptyRequest(t *testing.T) {
+	t.Parallel()
+
+	client := newHelloClient(t)
+
+	helloResp, err := client.SayHello(context.Background(), nil)
+	require.Error(t, err, "SayHello should return an error for nil request")
+	require.Empty(t, helloResp, "SayHello should return a nil response for a nil request")
+}
+
+func TestSayHello(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		name string
+
+		wantErr bool
+	}{
+		"Successfully greets a given name": {name: "Matt"},
+		"Error when name is empty":         {wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			client := newHelloClient(t)
+
+			helloReq := &provd.HelloRequest{
+				Name: tc.name,
+			}
+
+			helloResp, err := client.SayHello(context.Background(), helloReq)
+			if tc.wantErr {
+				require.Error(t, err, "SayHello should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "SayHello should not return an error, but did")
+
+			got := helloResp.GetMessage()
+			want := testutils.LoadWithUpdateFromGolden(t, got)
+			require.Equal(t, want, got, "GetAuthenticationModes returned an unexpected response")
+		})
+	}
+}
+
+// newHellolient returns a new GRPC Hello client for tests.
+func newHelloClient(t *testing.T) (client provd.HelloWorldServiceClient) {
+	t.Helper()
+
+	tmpDir, err := os.MkdirTemp("", "hello-socket-dir")
```

Comment:
My understanding is we don't have control over the rnd name that `t.Tempdir() ` creates, and unix sockets have a length limitation that it could hit, so we use this to control the socket path length?

I removed the comment by mistake, let me add it back in.

Reply:
Excellent! Oh, at least, I am consistent and there was a comment :p